rasa/nlu/extractors/mitie_entity_extractor.py

```diff
@@ -1,89 +1,130 @@
-from rasa.shared.nlu.constants import TEXT, ENTITIES
-from rasa.nlu.config import RasaNLUModelConfig
-from rasa.nlu.utils.mitie_utils import MitieNLP
-from rasa.nlu.tokenizers.tokenizer import Token, Tokenizer
-from rasa.nlu.components import Component
-from rasa.nlu.extractors.extractor import EntityExtractor
-from rasa.nlu.model import Metadata
+from rasa.shared.nlu.constants import (
+    ENTITY_ATTRIBUTE_CONFIDENCE,
+    ENTITY_ATTRIBUTE_START,
+    ENTITY_ATTRIBUTE_END,
+    ENTITY_ATTRIBUTE_TYPE,
+    ENTITY_ATTRIBUTE_VALUE,
+    TEXT,
+    ENTITIES,
+)
+from rasa.nlu.utils.mitie_utils import MitieModel
+from rasa.nlu.extractors.extractor import EntityExtractorMixin
 from rasa.shared.nlu.training_data.training_data import TrainingData
 from rasa.shared.nlu.training_data.message import Message
 import rasa.shared.utils.io
+from rasa.nlu.extractors._mitie_entity_extractor import MitieEntityExtractor
+from rasa.shared.exceptions import InvalidConfigException
 
 logger = logging.getLogger(__name__)
 
 if typing.TYPE_CHECKING:
     import mitie
 
+# TODO: remove when everything has been migrated
+MitieEntityExtractor = MitieEntityExtractor
 
-class MitieEntityExtractor(EntityExtractor):
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [MitieNLP, Tokenizer]
+
+class MitieEntityExtractorGraphComponent(GraphComponent, EntityExtractorMixin):
+    """A Mitie Entity Extractor (which is a thin wrapper around `Dlib-ml`)."""
+
+    MITIE_RESOURCE_FILE = "mitie_ner.dat"
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
+        return ["mitie"]
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """The component's default config (see parent class for full docstring)."""
+        return {
+            "num_threads": 1,
+        }
 
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
         ner: Optional["mitie.named_entity_extractor"] = None,
     ) -> None:
-        """Construct a new intent classifier using the sklearn framework."""
-        super().__init__(component_config)
-        self.ner = ner
+        """Creates a new instance.
+
+        Args:
+            config: The configuration.
+            model_storage: Storage which graph components can use to persist and load
+                themselves.
+            resource: Resource locator for this component which can be used to persist
+                and load itself from the `model_storage`.
+            ner: Mitie named entity extractor
+        """
+        # graph component
+        self._config = {**self.get_default_config(), **config}
+        self._model_storage = model_storage
+        self._resource = resource
+        self.validate_config(self._config)
+        # extractor
```

Comment:
Maybe a better named variable would make this comment redundant?

Reply:
i think "ner" is fine - the comments where just to sort out that there is stuff that a common graph component base class could do